### PR TITLE
fix(list, sortable-list, value-list): Emit calciteListOrderChange when dragging between lists

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7046

## Summary

TLDR: Basically, we need to pause draggable component connected/disconnected lifecycle events while a component is being dragged. Having these lifecycle methods kick off during a drag causes SortableJS errors.

- SortableComponent
  - Refactors logic to prevent any SortableJS component from doing its lifecycle logic when a component is being dragged
  - Previously, it was preventing all SortableJS components Sortable from being destroyed or created but that was causing issues by not emitting events when an item was moved from one list to another.
  - The nested component check that was previously being used isn't ideal because two different lists don't have to be nested to drag items between each other.
  - We need all lists to still continue emitting events when necessary, we just don't want their lifecycle methods to kick off when an item is being dragged. Otherwise, JS errors are thrown.
- Components
  - Updates SortableComponent components to not do any lifecycle callbacks when an item is being dragged to prevent any JS errors that SortableJS was throwing.
  - This was because in connectedCallback, sortable components were setting up the sortable instance, connecting the observer, modifying items, etc. We don't want the component to do this while an item is being dragged.  
  - The same thing as above was happening on disconnectedCallback.
  -  This fix stops all those errors that occurred while dragging an item from one list to another.

## Assumptions

It is reasonable to not do any lifecycle events for any draggable component while a component is being dragged

### CODE REVIEW

File: packages/calcite-components/src/utils/sortableComponent.ts [hunk 1]
```diff
@@ -59,14 +57,14 @@ export interface SortableComponent {
   onDragSort: (event: Sortable.SortableEvent) => void;
 
   /**
-   * Element dragging started.
+   * Called when any sortable component drag starts.
    */
-  onDragStart?: (event: Sortable.SortableEvent) => void;
+  onDragStart: () => void;
```
Comment (driskull) [inline packages/calcite-components/src/utils/sortableComponent.ts:62]: Should we rename these to something like `disableObservers`? Since that is what they are basically for.

File: packages/calcite-components/src/utils/sortableComponent.ts [hunk 2]
```diff
@@ -59,14 +57,14 @@ export interface SortableComponent {
...
 
   /**
-   * Element dragging ended.
+   * Called when any sortable component drag ends.
    */
-  onDragEnd?: (event: Sortable.SortableEvent) => void;
+  onDragEnd: () => void;
```
Comment (driskull) [inline packages/calcite-components/src/utils/sortableComponent.ts:67]: Should we rename these to something like `enableListeners`? Since that is what they are basically for.

File: packages/calcite-components/src/components/list/list.tsx [hunk 3]
```diff
@@ -477,6 +486,7 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
   }
 
   onDragSort(event: SortableEvent): void {
+    this.parentListEl = this.el.parentElement.closest("calcite-list");
```
Comment (driskull) [inline packages/calcite-components/src/components/list/list.tsx:489]: Since connectedCallback won't be fired, we should update the parentList here once dragging/sorting is done.

**Review by driskull — COMMENTED** [review]
questions

File: packages/calcite-components/src/components/list/list.e2e.ts [hunk 4]
```diff
@@ -471,6 +471,7 @@ describe("calcite-list", () => {
 
     it("works using a mouse", async () => {
       const page = await createSimpleList();
+      //const listOrderChangeSpy = await page.spyOnEvent("calciteListOrderChange");
```
Comment (driskull) [inline packages/calcite-components/src/components/list/list.e2e.ts:474]: @jcfranco I need your help with this one. If I add this spy, the test fails with the following...

```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down. From src/components/list/list.e2e.ts.

      at Object.get (../../node_modules/@stencil/core/testing/index.js:7:14)
      at Object.<anonymous> (../../node_modules/@stencil/core/testing/index.js:3772:79)

 RUNS  src/components/list/list.e2e.ts
/Users/dris0000/Sites/git/calcite-design-system/node_modules/@stencil/core/testing/index.js:3925
    throw new Error("The page was already closed");
          ^

Error: Caught error after test environment was torn down
```

Comment (jcfranco) [reply packages/calcite-components/src/components/list/list.e2e.ts]: Took a look and it looks like this error is caused by the drag event payload being serialized and there being circular JSON structures from the payload elements. I don't know yet what the proper fix for this would be, but you can work around it by manually adding an event listener to the page and tracking/storing the times it was emitted changes on `window` and then using that value to assert the final call count.

**Comment by driskull** [comment]
Test refactor issue: https://github.com/Esri/calcite-design-system/issues/7643

File: packages/calcite-components/src/components/list/list.e2e.ts (hunk 4 above)
Comment (driskull) [reply packages/calcite-components/src/components/list/list.e2e.ts]: Cool yeah that works for now. Thanks 👍

**Review by jcfranco — APPROVED** [review]
📃🤏➡️
